src/CommunityToolkit.Maui.Markup.SourceGenerators/SourceGenerators/TextAlignmentExtensionsGenerator.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CommunityToolkit.Maui.Markup.SourceGenerators;
+
+[Generator(LanguageNames.CSharp)]
+class TextAlignmentExtensionsGenerator : IIncrementalGenerator
+{
+	const string iTextAlignmentInterface = "Microsoft.Maui.ITextAlignment";
+	const string mauiControlsAssembly = "Microsoft.Maui.Controls";
+
+	public void Initialize(IncrementalGeneratorInitializationContext context)
+	{
+		// Get Microsoft.Maui.Controls Assembly Symbol
+		var mauiControlsAssemblySymbolProvider = context.CompilationProvider.Select(
+			static (compilation, token) => compilation.SourceModule.ReferencedAssemblySymbols.Single(q => q.Name == mauiControlsAssembly));
+
+		var inputs = mauiControlsAssemblySymbolProvider
+						.Combine(context.CompilationProvider)
+						.Select(static (combined, cancellationToken) => (MauiControlsAssemblySymbolProvider: combined.Left, Compilation: combined.Right));
+
+		context.RegisterSourceOutput(inputs, (context, collectedValues) =>
+			Execute(context, collectedValues.Compilation, collectedValues.MauiControlsAssemblySymbolProvider));
+	}
+
+	static void Execute(SourceProductionContext context, Compilation compilation, IAssemblySymbol mauiControlsAssemblySymbolProvider)
+	{
+		var textAlignmentSymbol = compilation.GetTypeByMetadataName(iTextAlignmentInterface);
+
+		if (textAlignmentSymbol is null)
+		{
+			var diag = Diagnostic.Create(TextAlignmentDiagnostics.MauiReferenceIsMissing, Location.None);
+			context.ReportDiagnostic(diag);
+			return;
+		}
+
+		var textAlignmentClassList = new List<(string ClassName, string ClassAcessModifier, string Namespace, string GenericArguments, string GenericConstraints)>();
```

Comment:
**Note for the future**

If the next SG that we implement uses the same types as this list, we should create a `record` for that. IMO that will be cleaner. If we don't use the same members again we can leave it as it's.